cpp/src/io/csv/writer_impl.cu

```diff
@@ -491,7 +510,8 @@ void write_csv(data_sink* out_sink,
           str_table_view.column(0), options_narep, stream, rmm::mr::get_current_device_resource());
       }();
 
-      write_chunked(out_sink, str_concat_col->view(), options, stream, mr);
+      write_chunked(
+        out_sink, str_concat_col->view(), options, stream, rmm::mr::get_current_device_resource());
```

Comment:
It makes sense to use the default device resource for writing data, but then why do we need to have an `mr` parameter above?

Reply:
Yes, I noticed that too. This function returns a void so there should actually be no need to pass a `mr` at all perhaps. That seemed like a bigger change out of scope of this PR so I was going to explore it in a different issue/PR. What do you think?

Reply:
Yes that sounds good. Thanks.